### IMPORT NECESSARY MODULES 

In [5]:
import pandas as pd
import scipy.stats as stats

### 1. A physician is evaluating a new diet for her patients with a family history of heart disease. To test the effectiveness of this diet, 16 patients are placed on the diet for 6 months. Their weights and triglyceride levels are measured before and after the study, and the physician wants to know if either set of measurements has changed. (Data set: dietstudy.csv) 

In [6]:
diet=pd.read_csv("D:/Analytics Labs(Data Science with Python)/Class 8 Files_Stats - 2 [Effect Size]/Stats - 2 [Effect Size]/Basic Statistics Assignment/dietstudy.csv")


In [11]:
diet.head()

patid  age  gender  tg0  tg1  tg2  tg3  tg4  wgt0  wgt1  wgt2  wgt3  wgt4
0      1   45    Male  180  148  106  113  100   198   196   193   188   192
1      2   56    Male  139   94  119   75   92   237   233   232   228   225
2      3   50    Male  152  185   86  149  118   233   231   229   228   226
3      4   46  Female  112  145  136  149   82   179   181   177   174   172
4      5   64    Male  156  104  157   79   97   219   217   215   213   214

* H0: NULL Hypothesis : mean of triglyceride before(tg0) = mean of triglyceride after(tg4)
                                                and 
                        mean of Weight before (wgt0) = mean of weight after (wgt4)
                        
* H1: Alternate Hypothesis : mean of triglyceride before(tg0) != mean of triglyceride after(tg4)
                                                OR
                             mean of Weight before (wgt0) != mean of weight after (wgt4)

* Confidence interval is 99% 
* Test : Paired Sample t-test on both  triglyceride and weight

In [8]:
stats.ttest_rel(a=diet.tg0,b=diet.tg4)

Ttest_relResult(statistic=1.2000008533342437, pvalue=0.24874946576903698)

* With 99% confidence interval we see that  pvalue(0.24) is not less than 1-Significance level(0.01) so NULL is accepted for  triglyceride 

In [14]:
stats.ttest_rel(a=diet.wgt0,b=diet.wgt4)

Ttest_relResult(statistic=11.174521688532522, pvalue=1.137689414996614e-08)

* With 99% confidence interval we see that pvalue is less than 1-Significance level(0.01) so NULL is rejected so we look at stats.

In [12]:
diet.wgt0.mean()

198.375

In [13]:
diet.wgt4.mean()

190.3125

* Mean of weight shows that weight has decreased after new diet.

### RESULT: With 99% confidence we reject the hypothesis that mean triglyceride level before is not equal to after and mean weight before is equal to mean weight after the new diet.

### 2. An analyst at a department store wants to evaluate a recent credit card promotion. To this end, 500 cardholders were randomly selected. Half received an ad promoting a reduced interest rate on purchases made over the next three months, and half received a standard seasonal ad. Is the promotion effective to increase sales? (Data set: creditpromo.csv) 

In [15]:
ccp=pd.read_csv("D:/Analytics Labs(Data Science with Python)/Class 8 Files_Stats - 2 [Effect Size]/Stats - 2 [Effect Size]/Basic Statistics Assignment/creditpromo.csv")

In [16]:
ccp.head()

id         insert      dollars
0   148       Standard  2232.771979
1   572  New Promotion  1403.807542
2   973       Standard  2327.092181
3  1096       Standard  1280.030541
4  1541  New Promotion  1513.563200

* H0: NULL Hypothesis : mean sales of standard promotion = mean sales of New promotion                                          
                                     
* H1: Alternate Hypothesis : mean sales of standard promotion != mean sales of New promotion 

* Confidence interval is 95% 
* Test : 2 Sample independent t-test

In [30]:
ccp_std=ccp[ccp['insert']=='Standard'].dollars

In [31]:
ccp_new=ccp[ccp['insert']=='New Promotion'].dollars

In [32]:
stats.ttest_ind(a=ccp_std,b=ccp_new,equal_var=False)

Ttest_indResult(statistic=-2.260422726464996, pvalue=0.024226348191648994)

With 95% confidence interval we see that pvalue(0.024) < 1-significance level(0.05) so WE REJECT THE NULL and see the stats

In [33]:
ccp_std.mean()

1566.3890309659348

In [35]:
ccp_new.mean()

1637.4999830647992

Mean values shows that sales have increased after New promotion 

### RESULT: With 95% Confidence interval Mean sales before Promotion is not equal to Mean sales after Promotion and accoding to data Mean sales have increased after new sales Promotion.

### 3. An experiment is conducted to study the hybrid seed production of bottle gourd under open field conditions. The main aim of the investigation is to compare natural pollination and hand pollination. The data are collected on 10 randomly selected plants from each of natural pollination and hand pollination. The data are collected on fruit weight (kg), seed yield/plant (g) and seedling length (cm). (Data set: pollination.csv)
* a. Is the overall population of Seed yield/plant (g) equals to 200?
* b. Test whether the natural pollination and hand pollination under open field conditions are equally effective or are significantly different

In [36]:
pollination=pd.read_csv("D:/Analytics Labs(Data Science with Python)/Class 8 Files_Stats - 2 [Effect Size]/Stats - 2 [Effect Size]/Basic Statistics Assignment/pollination.csv")

In [40]:
pollination.head()

Group  Fruit_Wt  Seed_Yield_Plant  Seedling_length
0  Natural      1.85            147.70            16.86
1  Natural      1.86            136.86            16.77
2  Natural      1.83            149.97            16.35
3  Natural      1.89            172.33            18.26
4  Natural      1.80            144.46            17.90

a)

* H0: NULL Hypothesis : Seed yield/plant == 200                                       
                                     
* H1: Alternate Hypothesis : Seed yield/plant != 200 

* Confidence interval is 95% 
* Test : 1 Sample independent t-test

In [45]:
stats.ttest_1samp(a=pollination.Seed_Yield_Plant,popmean=200)

Ttest_1sampResult(statistic=-2.3009121248548645, pvalue=0.032891040921283025)

With 95% confidence interval we see that pvalue(0.032) < 1-significance level(0.05) so WE REJECT THE NULL and see the stats

In [47]:
pollination.Seed_Yield_Plant.mean()

180.8035

Mean value shows that Seed yeild/plant is less than 200

### RESULT: With 95% Confidence interval seed yeild/plant  is not equal to 200 and accoding to data seed yeild/plant is less than 200

b)
* H0: Mean of Fruit Wt(by Hand Pollination) = Mean of Fruit Wt(By Natural Pollination)
                                  and 
      Mean of Seed_Yield_Plant(by Hand Pollination) = Mean of Seed_Yield_Plant(By Natural Pollination)
                                  and 
      Mean of Seedling Length(by Hand Pollination) = Mean of Seedling Length(By Natural Pollination)                              
* H1: Mean of Fruit Wt(by Hand Pollination) != Mean of Fruit Wt(By Natural Pollination)
                                  and 
      Mean of Seed_Yield_Plant(by Hand Pollination) != Mean of Seed_Yield_Plant(By Natural Pollination)
                                  and 
     Mean of Seedling Length(by Hand Pollination)!= Mean of Seedling Length(By Natural Pollination)                              

* Confidence interval is 95% 
* Test : 2 Sample paired  t-test on multiple data set

In [48]:
pollination.head(3)

Group  Fruit_Wt  Seed_Yield_Plant  Seedling_length
0  Natural      1.85            147.70            16.86
1  Natural      1.86            136.86            16.77
2  Natural      1.83            149.97            16.35

In [52]:
nat_fruitwt=pollination[pollination.Group=='Natural'].Fruit_Wt

In [54]:
hand_fruitwt=pollination[pollination.Group=='Hand'].Fruit_Wt

In [55]:
stats.ttest_rel(a=nat_fruitwt,b=hand_fruitwt)

Ttest_relResult(statistic=-16.003484109505287, pvalue=6.420421224715851e-08)

With 99% confidence interval we see that pvalue < 1-significance level(1-0.99=0.01) so WE REJECT THE NULL and see the stats

In [80]:
pollination[pollination.Group=='Natural'].Fruit_Wt.mean()

1.848

In [81]:
pollination[pollination.Group=='Hand'].Fruit_Wt.mean()

2.5660000000000003

According to stats mean of Fruit Wt. by Hand(2.56) is greater than mean of fruit wt. by natural pollination(1.84)

In [70]:
nat_syp=pollination[pollination.Group=='Natural'].Seed_Yield_Plant

In [72]:
hand_syp=pollination[pollination.Group=='Hand'].Seed_Yield_Plant

In [73]:
stats.ttest_rel(a=nat_syp,b=hand_syp)

Ttest_relResult(statistic=-16.80079294985747, pvalue=4.198813240596638e-08)

In [82]:
pollination[pollination.Group=='Natural'].Seed_Yield_Plant.mean()

146.009

In [83]:
pollination[pollination.Group=='Hand'].Seed_Yield_Plant.mean()

215.598

According to stats mean of Seed_yeild_plant by Hand(215.5) is greater than mean of Seed_yeild_plant by natural pollination(146)

In [75]:
nat_sl=pollination[pollination.Group=='Natural'].Seedling_length

In [76]:
hand_sl=pollination[pollination.Group=='Hand'].Seedling_length

In [77]:
stats.ttest_rel(a=nat_sl,b=hand_sl)

Ttest_relResult(statistic=-2.9765905136151303, pvalue=0.015535315669362245)

In [78]:
pollination[pollination.Group=='Natural'].Seedling_length.mean()

17.706999999999994

In [79]:
pollination[pollination.Group=='Hand'].Seedling_length.mean()

18.589999999999996

According to stats mean of Seedling_length by Hand(17.70) is greater than mean of Seedling_length by natural pollination(18.58)

### RESULT: With 99% Confidence interval Natural  Pollination and Hand Pollination under open field conditions are not equally effective and according to statistics Hand pollination > Natural pollination


### 4. An electronics firm is developing a new DVD player in response to customer requests. Using a prototype, the marketing team has collected focus data for different age groups viz. Under 25; 25-34; 35-44; 45-54; 55-64; 65 and above. Do you think that consumers of various ages rated the design differently? (Data set: dvdplayer.csv). 

In [85]:
dvd=pd.read_csv("D:/Analytics Labs(Data Science with Python)/Class 8 Files_Stats - 2 [Effect Size]/Stats - 2 [Effect Size]/Basic Statistics Assignment/dvdplayer.csv")

In [96]:
dvd.head()

agegroup   dvdscore
0  65 and over  38.454803
1        55-64  17.669677
2  65 and over  31.704307
3  65 and over  25.924460
4     Under 25  30.450007

* H0: mean dvd_un25 = dvd_25_34 = dvd_35_44 = dvd_45_54 = dvd_55_64 = dvd_65_over

* H1: atleast one of the mean is not equal to other                           

* Confidence interval is 99% 
* Test : ANOVA 

In [88]:
dvd_un25=dvd.dvdscore[dvd.agegroup=='Under 25']

In [90]:
dvd_25_34=dvd.dvdscore[dvd.agegroup=='25-34']

In [91]:
dvd_35_44=dvd.dvdscore[dvd.agegroup=='35-44']

In [92]:
dvd_45_54=dvd.dvdscore[dvd.agegroup=='45-54']

In [93]:
dvd_55_64=dvd.dvdscore[dvd.agegroup=='55-64']

In [94]:
dvd_65_over=dvd.dvdscore[dvd.agegroup=='65 and over']

In [95]:
stats.f_oneway(dvd_un25,dvd_25_34,dvd_35_44,dvd_45_54,dvd_55_64,dvd_65_over)

F_onewayResult(statistic=6.992526962676518, pvalue=3.087324905679639e-05)

In [97]:
dvd_un25.mean()

28.749228425584857

In [98]:
dvd_25_34.mean()

31.67800994228487

In [99]:
dvd_35_44.mean()

37.018058228022774

In [100]:
dvd_45_54.mean()

39.1183687318563

In [101]:
dvd_55_64.mean()

28.447335692086785

In [102]:
dvd_65_over.mean()

28.00279130552502

### RESULT: With 99% Confidence interval  we can say that consumers of various ages rated the design differently    and data also depicts the same